src/winrpath.cxx

```diff
@@ -0,0 +1,1025 @@
+#pragma once
+#include "winrpath.h"
+#include "utils.h"
+
+#include <sstream>
+
+#define BUFSIZE 4096
+
+using CoffMembers = std::vector<coff_entry>;
+
+
+const std::map<char, char> special_character_to_path{
+    {'|', '\\'},
+    {';', ':'}
+};
+
+const std::map<char, char> path_to_special_characters{
+    {'\\', '|'},
+    {'/', '|'},
+    {':', ';'}
+};
+
+/**
+ * Replaces characters used to mangle path characters with
+ * valid path characters
+ * 
+ * \param in a pointer to the string to replace the mangled path characters in
+ * \param len the length of the mangled path
+ */
+void replace_special_characters(char in[], int len)
+{
+    for (int i = 0; i < len; ++i) {
+        if (special_character_to_path.count(in[i]))
+            in[i] = special_character_to_path.at(in[i]);
+    }
+}
+
+/**
+ * Replaces path characters with special, non path, replacement characters
+ * 
+ * \param in a pointer to the string to have its path characters replace with special placeholders
+ * \param len the length of the path to be mangled
+ */
+void replace_path_characters(char in[], int len)
+{
+    for (int i = 0; i < len; i++ ) {
+        if (path_to_special_characters.count(in[i]))
+            in[i] = path_to_special_characters.at(in[i]);
+    }
+}
+
+/**
+ * Pads a given path with an amount of padding of special characters
+ * 
+ * \param pth a pointer to the path to be padded
+ * \param str_size the length of the path
+ * \param bsize the lengh of the padding to add
+ */
+char * pad_path(const char *pth, DWORD str_size, DWORD bsize = MAX_PATH)
```

Comment:
(question) this is not currently used: is that going to change?

Reply:
Offline discussion summary:
We don't actually need to pad because we can re-run the librarian tool (`lib.exe`) on buildcache extraction, so no padding needed and due to the name length limit, the use of padding really only hurts us.